functions/self_update.go

```diff
@@ -0,0 +1,43 @@
+package functions
+
+import (
+	"log"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/gravitl/netclient/daemon"
+	"github.com/rhysd/go-github-selfupdate/selfupdate"
+)
+
+// SelfUpdate updates the netclient binary in place to the latest release available on GitHub
+// and reboots the daemon if update is successful
+// All binary names must adhere to the format `netclient-{platform}-{architecture}`
+func SelfUpdate(currentVersion string) {
+	if currentVersion == "dev" {
+		return
+	}
+	semVer := strings.Replace(currentVersion, "v", "", -1)
```

Comment:
is this going to handle a version like v0.16.3-testing?  
you probably want to do something like this
```
	trimmed := strings.TrimFunc(ver, func(r rune) bool {
		return !unicode.IsNumber(r)
	})
```

Reply:
done